packages/git/src/node/dugite-git.ts

```diff
@@ -657,13 +657,37 @@ export class DugiteGit implements Git {
             const file = Path.relative(this.getFsPath(repository), this.getFsPath(options.uri)) || '.';
             args.push(...[file]);
         }
-        const result = await this.exec(repository, args);
+
+        const successExitCodes = new Set([0, 128]);
+        let result = await this.exec(repository, args, { successExitCodes });
+        if (result.exitCode !== 0) {
+            // The 'log' command could potentially be valid when no HEAD if the revision range does not involve HEAD */
+            if ((!options || !options.range || options.range.toRevision === 'HEAD') && !await this.revParse(repository, { ref: 'HEAD' })) {
```

Comment:
With the way you wrote it, if `options` is undefined, the `revParse` test will be executed, is this on purpose? Is git relying on HEAD when you don't specify anything?

Reply:
Yes, you are correct that the default is 'HEAD' so I believe the code is correct.  I have refactored a little to make the code less confusing.